Replicator/Replicator.cc

```diff
@@ -241,11 +246,14 @@ namespace litecore { namespace repl {
 
     void Replicator::onError(C4Error error) {
         Worker::onError(error);
-        if (error.domain == LiteCoreDomain && error.code == kC4ErrorUnexpectedError) {
-            // Treat an exception as a fatal error for replication:
-            alloc_slice message( c4error_getDescription(error) );
-            logError("Stopping due to fatal error: %.*s", SPLAT(message));
-            _disconnect(websocket::kCodeUnexpectedCondition, "An exception was thrown"_sl);
+        for(const C4Error& stoppingErr : StoppingErrors) {
+            if(stoppingErr.domain == error.domain && stoppingErr.code == error.code) {
+                // Treat an exception as a fatal error for replication:
+                alloc_slice message( c4error_getDescription(error) );
+                logError("Stopping due to fatal error: %.*s", SPLAT(message));
```

Comment:
I'd rather not report these the same way, since they're quite different conditions. How about just adding an `else` condition after the existing `if` to check for 503, and handle it by logging a warning and setting the disconnect message to "The server is over capacity".

Reply:
You mean replace the message received from the other side with a hardcoded one?  I think it might be a good idea to keep the original around.  Although I could make a distinction between "fatal" and not "fatal"